src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -263,15 +263,23 @@ protected function toNumber($value)
         if (is_numeric($value)) {
             return $value + 0; // cast to number
         }
+        if (is_bool($value) || is_null($value)) {
+            return (int) $value;
+        }
+        if ($this->getTypeCheck()->isArray($value) && count($value) === 1) {
+            reset($value);
```

Comment:
`reset` returns the first element in the array, so you can skip a step and  pass its results directly on line 272